src/transaction.ts

```diff
@@ -458,3 +492,215 @@ export class TransactionHash extends Hash {
+      receiver: string;
+      senderUsername?: string;
+      receiverUsername?: string;
+      gasPrice?: BigNumber.Value;
+      gasLimit: BigNumber.Value;
+      data?: Uint8Array;
+      chainID: string;
+      version?: number;
+      options?: number;
+      guardian?: string;
+    }) {
+      this.nonce = nonce || 0;
+      this.value = value || new BigNumber(0);
+      this.sender = sender;
+      this.receiver = receiver;
+      this.senderUsername = senderUsername || "";
+      this.receiverUsername = receiverUsername || "";
+      this.gasPrice = gasPrice || new BigNumber(TRANSACTION_MIN_GAS_PRICE);
+      this.gasLimit = gasLimit;
+      this.data = data || new Uint8Array();
+      this.chainID = chainID;
+      this.version = version || TRANSACTION_OPTIONS_DEFAULT;
+      this.options = options || TRANSACTION_OPTIONS_DEFAULT;
+      this.guardian = guardian || "";
+  
+      this.signature = new Uint8Array();
+      this.guardianSignature = new Uint8Array();
+    }
+}
+
+/**
+ * An utilitary class meant to work together with the {@link TransactionNext} class.
+ */
+export class TransactionComputer {
+  constructor() {}
+
+  computeTransactionFee(transaction: ITransactionNext, networkConfig: INetworkConfig): BigNumber{
+    let moveBalanceGas = new BigNumber(
+      networkConfig.MinGasLimit + transaction.data.length * networkConfig.GasPerDataByte);
+    if (moveBalanceGas > transaction.gasLimit) {
+      throw new errors.ErrNotEnoughGas(parseInt(transaction.gasLimit.toString(), 10));
+    }
+
+    let gasPrice = new BigNumber(transaction.gasPrice);
+    let feeForMove = moveBalanceGas.multipliedBy(gasPrice);
+    if (moveBalanceGas === transaction.gasLimit) {
+      return feeForMove;
+    }
+
+    let diff = new BigNumber(transaction.gasLimit).minus(moveBalanceGas);
+    let modifiedGasPrice = gasPrice.multipliedBy(
+      new BigNumber(networkConfig.GasPriceModifier)
+    );
+    let processingFee = diff.multipliedBy(modifiedGasPrice);
+
+    return feeForMove.plus(processingFee);
+  }
+
+  computeBytesForSigning(transaction: ITransactionNext): Uint8Array{
+    let plainTransaction = this.toPlainObject(transaction);
+
+    if (plainTransaction.signature) {
+      delete plainTransaction.signature;
+    }
+
+    if (plainTransaction.guardianSignature) {
+      delete plainTransaction.guardianSignature;
+    }
+
+    if (!plainTransaction.guardian) {
+      delete plainTransaction.guardian
+    }
+
+    const serialized = JSON.stringify(plainTransaction);
+
+    return new Uint8Array(Buffer.from(serialized));
+  }
+
+  computeTransactionHash(transaction: ITransactionNext): Uint8Array{
+    let serializer = new ProtoSerializer();
+
+    const tx = Transaction.fromTransactionNext(transaction);
+    let buffer = serializer.serializeTransaction(tx);
+    let hash = createTransactionHasher(TRANSACTION_HASH_LENGTH)
+      .update(buffer)
+      .digest("hex");
+    
+    return new Uint8Array(Buffer.from(hash, "hex"));
```

Comment:
Generally speaking, the following:

```
return new Uint8Array(Buffer.from(hash, "hex"))
```

Can be replaced by:

```
return Buffer.from(hash, "hex")
```

(optional, shorter)

Reply:
Indeed, replaced it.